README.md

```diff
@@ -771,10 +771,10 @@ App\Entity\User:
         gender:
             - Elao\Enum\Bridge\Symfony\Validator\Constraint\Enum:
                 class: MyApp\Enum\Gender
-                callback: ['allowedValues']
+                callback: 'allowedValues'
 ```
 
-Where `allowedValues` is a static method of `MyApp\Enum\Gender`, returning allowed instances (:warning: should return values if `asValue` is set to `true`).
+Where `allowedValues` is a static method of `MyApp\Enum\Gender`, returning allowed values or instances.
```

Comment:
Actually the documentation seems a bit out of sync with the code to me.

There is no check for the method name: the callback is called directly. So the method could be named anything. Maybe we could keep `allowedValues` in the example but remove it from the documentation text.

And `asValue` is only checked when there is no user callback, so the user can control whether he wants to get values or instances. It makes sense to me because otherwise the user can just implement the callback to return whatever he wants.

Reply:
This is just an example actually :) Of course it can be named as you want.

> And asValue is only checked when there is no user callback

Yes but if `asValue` is set to true, then the callback set by the user must return values, not instances. That's the reason behind this note in the example.
Or did I misunderstand you?

Reply:
No that's what I meant, but I still don't get why the user would use `asValue` when using his own callback since he can already choose to return as value or as instances. As I understand it, `asValue` is not used at all in case of a user callback.

Reply:
Alright, makes sense